pandas/core/internals/construction.py

```diff
@@ -615,6 +615,8 @@ def _extract_index(data) -> Index:
             elif is_list_like(val) and getattr(val, "ndim", 1) == 1:
                 have_raw_arrays = True
                 raw_lengths.append(len(val))
+            elif isinstance(val, np.ndarray) and val.ndim > 1:
+                raise ValueError("Data must be 1-dimensional")
```

Comment:
maybe something like "Per-column arrays must each be 1-dimensional"?

Reply:
amended